addons/website_sale/controllers/main.py

```diff
@@ -145,17 +145,36 @@ def _get_shop_domain(self, search, category, attrib_values, search_in_descriptio
         return expression.AND(domains)
 
     def sitemap_shop(env, rule, qs):
+        website = env['website'].get_current_website()
+        # Make sure urls are not listed in sitemap when restriction is active
+        if website and website.ecommerce_access == 'logged_in':
+            return
+
         if not qs or qs.lower() in '/shop':
             yield {'loc': '/shop'}
 
         Category = env['product.public.category']
         dom = sitemap_qs2dom(qs, '/shop/category', Category._rec_name)
-        dom += env['website'].get_current_website().website_domain()
+        dom += website.website_domain()
         for cat in Category.search(dom):
             loc = '/shop/category/%s' % slug(cat)
             if not qs or qs.lower() in loc:
                 yield {'loc': loc}
 
+    def sitemap_products(env, rule, qs):
+        website = env['website'].get_current_website()
+        # Make sure urls are not listed in sitemap when restriction is active
+        if website and website.ecommerce_access == 'logged_in':
+            return
```

Comment:
FYI, it's not what you want as the `sitemap=` in controllers is also used to autocomplete link creation for admins etc.
Here, the product pages are not shown anymore to the admin when creating a link, which you still want to show even if the eshop is restricted:
- Admin could create a link on a restricted page access, or simply from one product page to another one
- Admin could create a link in a mass mailing campaign etc.
- (more advanced) Admin could create a menu (restricted to a group) linked to the product

To block it from sitemap, you could do that in a `if not qs:` condition. If there is a `qs`, it means it's in the autocomplete case basically.

Reply:
Oh, good to know, will update that.
Hesitated to ask you for review, but since you didn't work in website anymore, didn't want to annoy you :D (and looked quite safe by matching the existing sitemap methods).

Reply:
Hm, you could TBH, I would just dispatch to someone else if I know someone can answer correctly and if am too busy here in PS Tech :p 

I've always said to my team (or anyone bring the subject) that there is never too many pings, worst case you just explain that you have no time and to move on without your review or to dispatch to someone else

Note: I was just looking for a commit on the saas-17.4 that might have break something in accounting (nothing at all related to website_*) and I just got curious about this one when I saw the title, in case you wonder :innocent: